src/bot/integrations/discord.ts

```diff
@@ -98,6 +98,20 @@ class Discord extends Integration {
   })
   sendGeneralAnnounceToChannel = '';
 
+  @settings('bot')
+  @ui({
+    type: 'selector', values: ['online', 'invisible', 'dnd'],
+    if: () => self.guild.length > 0,
+  })
+  onlinePresence: 'online' | 'idle' | 'invisible' | 'dnd' = 'online';
```

Comment:
https://github.com/sogehige/sogeBot/pull/4030#discussion_r470027846
https://github.com/sogehige/sogeBot/pull/4030#discussion_r470028519

Reply:
> also this seem to be confusing to me as well...we should have configurable online presense for offline/online stream states. Also we should have configurable activity as well (maybe just toggle to set streaming activity during online stream) -> thus keeping old behavior

if toggle is enabled (default) bot change its status to streaming if stream goes live
if togge is disabled bot stay in selected status like bot actualy do

> so we should have
this.onlinePresence -> presence when stream is online
this.offlinePresence -> presence when stream is offline

i will add this if i can

and i request onHosting() -> set bot to watching status
as a third option